### PR TITLE
Fix Client URL Prepending For GraphQL Endpoint on Enterprise

### LINKED ISSUE #1047: Fix Client URL Prepending For GraphQL Endpoint on Enterprise
We noticed a potential bug when trying to use the library with the V4/GraphQL API and our Enterprise server.

The URL prepending that occurs in the Client class for GHE always sets the prefix to `/api/{version}` but for the GraphQL/v4 endpoint the prepend needs to be `/api/graphql`.   This leads to a `Not Found Exception` on all queries.  The class defaults to `v3` if no version is provided so we can't just send a blank to get `/api`

```
$builder->addPlugin(new PathPrepend(sprintf('/api/%s', $this->getApiVersion())));
```
results in a call to `https://{enterprise_url}/api/v3/graphql` which is not the real endpoint so you end up with a not found exception.

```
  Uncaught Github\Exception\RuntimeException: Not Found
```

### PR DESCRIPTION
**Pull request by asher-goldberg**
fixes #1047 

This checks if `v4` is passed into the Client and an Enterprise URL is set then the prepend will only be `/api`.  (The class default is `v3` if no version is provided so can't just pass an empty string).

TL;DR - this now allows this package to run GraphQL calls on Enterprise `$client = new Client(null, 'v4', 'https://enterpriseurl');`

### CODE REVIEW

**Comment by acrobat** [comment]
Thanks @asher-goldberg! And congrats on your first contribution! 🎉
